packages/rest/src/rest.ts

```diff
@@ -444,15 +463,26 @@ export function $action(context: DecoratorContext, entity: Operation, name?: str
     });
     return;
   }
+
   const action = makeActionName(entity, name);
-  context.call($actionSegment, entity, action);
+  context.call($actionSegment, entity, action.name);
   context.program.stateMap(actionsKey).set(entity, action);
 }
 
-export function getAction(program: Program, operation: Operation): string | null | undefined {
+/**
+ * Gets the ActionDetails for the specified operation if it has previously been marked with @action.
+ */
+export function getActionDetails(
+  program: Program,
+  operation: Operation
+): ActionDetails | undefined {
   return program.stateMap(actionsKey).get(operation);
 }
 
+export function getAction(program: Program, operation: Operation): string | null | undefined {
+  return getActionDetails(program, operation)?.name;
+}
+
```

Comment:
Should we deprecate this in favor of `getActionDetails`?

Reply:
We could, I just didn't want to disrupt existing code.  I'm fine with throwing a deprecation on it if others agree.

Reply:
we shouldn't throw deprecation in getters or we end up with diagnostic emitted all the time that are very confusing just because an emitter didn't update.

But we should add `@deprecated` in the function js doc

Reply:
It's similar to what we had with the bad versioning helpers, where we kept them in, just returning the first value, but we deprecated them because we ultimately wanted people to move to the right pattern. I'd +1 @timotheeguerin's suggestion.

Reply:
OK, looks like `getAction` is used only in the ARM controller generator and `getCollectionAction` isn't used anywhere, so we might be able to deprecate these two functions in favor of the `*Details` variant.  I'll go ahead and do that.

Reply:
Yeah, I was planning to use jsdoc style for that, not diagnostics.